### PR TITLE
UserTokenListAPIHandler support Authenticator.authenticate returning a dict

### PR DESCRIPTION
**Pull request by gweis**
The documentation for `Authenticator.authenticate` allows returning a dictionary with a mandatory key `name`.

The `authenticate` method call in `UserTokenListAPIHandler.post` should support this as well.

(Included test fails without the patch).

### CODE REVIEW

File: jupyterhub/apihandlers/users.py [hunk 1]
```diff
@@ -243,6 +243,9 @@ def sort_key(token):
             # can be username+password or an upstream auth token
             try:
                 name = await self.authenticator.authenticate(self, body.get('auth'))
```
[the remark this answers is not in the document]
Comment (gweis) [reply jupyterhub/apihandlers/users.py]: I was not sure whether this was desired, as self.auth_to_user also creates the the user in the database if it does not exist already, and with OIDC login, you might not get a full auth_state (e.g. refresh token), from the authenticator.

[the remark this answers is not in the document]
Comment (minrk) [reply jupyterhub/apihandlers/users.py]: This is meant to behave the same as a POST to the login form, so calling the same code as is used there should be the right thing to do.

[the remark this answers is not in the document]
Comment (gweis) [reply jupyterhub/apihandlers/users.py]: I am rather thinking about the use case discussed in #1840, where it is possible to use an external access token, to mint a jupyterhub token. This does not really fit into the pattern with posting username / password or an already existing jupyterhub token. I am probably fine with calling `self.auth_to_user` here, but would like to double check with you and the reference to #1840 whether that's really what should happen here.

Partly addressed comments by calling handlers `self.authenticate` as suggested.

**Comment by minrk** [comment]
Thanks!

**Pull request merged**